node/node.go

```diff
@@ -326,6 +326,19 @@ func (n *Node) Stop(ctx context.Context) error {
 	default:
 		return errNodeNotStarted
 	}
+	// TODO(dongluochen): node can coordinate with manager to remove assignments.
+	// a node sends a 'leave' message to manager (thru heartbeat?), waits for
+	// manager to drain the node, and stops. if manager is not responding, it
```

Comment:
Seems unnecessary. We could have simply a final "Leave message", that simply removes the node from the nodes table, and then rely on the local cleanup.